iot-hub/Samples/device/EdgeDownstreamDevice/Program.cs

```diff
@@ -0,0 +1,131 @@
+        //
+        // Either set the DEVICE_CONNECTION_STRING environment variable with this connection string
+        // or set it in the launchSettings.json.
+        private static string DeviceConnectionString = Environment.GetEnvironmentVariable("DEVICE_CONNECTION_STRING");
+
+        private static int MESSAGE_COUNT = 10;
+        private const int TEMPERATURE_THRESHOLD = 30;
+
+        private static float temperature;
+        private static float humidity;
+
+        /// <summary>
+        /// First install any CA certificate provided by the user to connect to the Edge device.
+        /// Next attempt to connect to the Edge device and send it DEVICE_MESSAGE_COUNT
+        /// number of telemetry data messages.
+        ///
+        /// Note: Either set the DEVICE_MESSAGE_COUNT environment variable with the number of
+        /// messages to be sent to the IoT Edge runtime or set it in the launchSettings.json.
+        /// </summary>
+        static void Main()
+        {
+            InstallCACert();
+
+            try
+            {
+                MESSAGE_COUNT = Int32.Parse(Environment.GetEnvironmentVariable("DEVICE_MESSAGE_COUNT"), NumberStyles.None, new CultureInfo("en-US"));
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Invalid number of messages in env variable DEVICE_MESSAGE_COUNT. MESSAGE_COUNT set to {0}\n", MESSAGE_COUNT);
+            }
+
+            DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(DeviceConnectionString);
+
+            if (deviceClient == null)
+            {
+                Console.WriteLine("Failed to create DeviceClient!");
+            }
+            else
+            {
+                SendEvents(deviceClient).Wait();
+            }
+
+            Console.WriteLine("Exiting!\n");
+        }
+
+        /// <summary>
+        /// Add certificate in local cert store for use by downstream device
+        /// client for secure connection to IoT Edge runtime.
+        ///
+        ///    Note: On Windows machines, if you have not run this from an Administrator prompt,
+        ///    a prompt will likely come up to confirm the installation of the certificate.
+        ///    This usually happens the first time a certificate will be installed.
+        /// </summary>
+        static void InstallCACert()
+        {
+            string certPath = Environment.GetEnvironmentVariable("EDGE_DEVICE_CA_CERTIFICATE_PATH");
+            if (!string.IsNullOrWhiteSpace(certPath))
+            {
+                Console.WriteLine("User configured CA certificate path: {0}", certPath);
+                if (!File.Exists(certPath))
+                {
+                    // cannot proceed further without a proper cert file
+                    Console.WriteLine("Invalid certificate file: {0}", certPath);
+                    throw new InvalidOperationException("Invalid certificate file.");
```

Comment:
`FileNotFoundException` - the file doesn't exist, it's not that the file exists and it's invalid.